pkg/p2p/message_router.go

```diff
@@ -55,20 +57,32 @@ type messageRouterImpl struct {
 	credentials  *security.Credential
 	selfID       NodeID
 	clientConfig *MessageClientConfig
+
+	enableLocalClient bool
 }
 
-// NewMessageRouter creates a new MessageRouter
-func NewMessageRouter(selfID NodeID, credentials *security.Credential, clientConfig *MessageClientConfig) MessageRouter {
+func NewMessageRouterWithLocalClient(
+	selfID NodeID,
+	credentials *security.Credential,
+	clientConfig *MessageClientConfig,
+	enableLocalClient bool,
```

Comment:
`NewMessageRouterWithLocalClient` mean we always have a local client, why `enableLocalClient` is need?  I think that's very confusing.

Reply:
fixed, ptal!